src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -122,8 +122,22 @@ def create_metric_alert(
         find_channel_id_for_alert_rule.apply_async(kwargs=task_args)
         return Response({"uuid": client.uuid}, status=202)
     else:
-        alert_rule = serializer.save()
-        return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
+        alert_rule = validator.save()
+        if features.has("organizations:workflow-engine-rule-serializers", organization):
+            try:
+                detector = Detector.objects.get(alertruledetector__alert_rule_id=alert_rule.id)
+                return Response(
+                    serialize(
+                        detector,
+                        request.user,
+                        WorkflowEngineDetectorSerializer(),
+                    ),
+                    status=status.HTTP_201_CREATED,
+                )
+            except Detector.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+        else:
+            return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
```

Comment:
totally stylistic and extreme nit, but you don't need the `else:` block if all the paths in the `if` block will return
```suggestion
        return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
```